# 2つの互いに相関がある資産

## (a)

資産$A$の期待収益を$\bar r_A$、資産$B$の期待収益を$\bar r_B$とする。
\begin{align}
\bar r_A = 0.10,\ \sigma_A = 0.15\\
\bar r_b = 0.18,\ \sigma_B = 0.30
\end{align}

ポートフォリオの期待収益を$\bar r$、その標準偏差を$\sigma$とすると、
\begin{align}
\bar r &= \alpha\bar r_A + (1-\alpha)\bar r_B\\
\sigma^2 &= \alpha^2\sigma_A^2 + 2\alpha(1-\alpha)\sigma_{AB} + (1-\alpha)^2\sigma_B^2
\end{align}

$A$と$B$の相関係数$\rho=0.1$より、
\begin{align}
\sigma_{AB} &= \rho\sigma_A\sigma_B\\
\\
&= 0.1\times0.15\times0.30\\
\\
&= 0.0045
\end{align}

よって
\begin{align}
\sigma^2 &= 0.15^2\alpha^2 + 2\times0.0045\times\alpha(1-\alpha) + 0.30^2(1-\alpha)^2\\
\\
&= 0.1035\alpha^2 - 0.171\alpha + 0.09\\
\\
&= 0.1035(\alpha - \frac{19}{23})^2+\mbox{Const.}
\end{align}
より、ポートフォリオの標準偏差を最小化する$\alpha$は<strong>$\frac{19}{23}$</strong>である。

## (b)

\begin{align}
\sigma &= \sqrt{0.1035\alpha^2 - 0.171\alpha + 0.09}\\
\\
&= \sqrt{0.1035\times\left(\frac{19}{23}\right)^2 - 0.171\times\frac{19}{23} + 0.09}\\
\\
&= 0.13917...
\end{align}

In [14]:
import math
math.sqrt(0.1035*(19/23)**2-0.171*19/23+0.09)

0.13917458538609448

よって最小標準偏差は$13.9\%$

## (c)

\begin{align}
\bar r &= \alpha\bar r_A + (1-\alpha)\bar r_B\\
\\
&= \frac{19}{23}\times0.10 + (1-\frac{19}{23})\times0.18\\
\\
&= \frac{2.62}{23} = 0.1139...
\end{align}

よってポートフォリオの期待収益率は$11.4\%$